RELEASING.md

```diff
@@ -0,0 +1,81 @@
+# Releasing Testcontainers for Go
+
+In order to create a release, we have added a shell script that performs all the tasks for you, allowing a dry-run mode for checking it before creating the release. We are going to explain how to use it in this document.
+
+First, it's really important that you first check that the [version.go](./internal/version.go) file is up-to-date, containing the right version you want to create. That file will be used by the automation to perform the release.
+Once the version file is correct in the repository:
+
+- Run the [release.sh](./scripts/release.sh) shell script to run it in dry-run mode.
+- You can run the script without dry-run setting `DRY_RUN=false` in the environment:
+
+        DRY_RUN="false" ./scripts/release.sh
+
+- The script will create a git tag with the current value of the [version.go](./internal/version.go) file, starting with `v`: e.g. `v0.18.0`, for the following Go modules:
+    - the root module, representing the Testcontainers for Go library.
+    - all the Go modules living in both the `examples` and `modules` directory. The git tag value for these Go modules will be created using this name convention:
+
+             "${directory}/${module_name}/${version}", e.g. "examples/mysql/v0.18.0", "modules/compose/v0.18.0"
+
+- The script will update the [mkdocs.yml](./mkdocks.yml) file, updating the `latest_version` field to the current version.
+- The script will update the [version.go](./internal/version.go) file, setting the next development version to the next **minor** version by default. For example, if the current version is `v0.18.0`, the script will update the [version.go](./internal/version.go) file with the next development version `v0.19.0`.
+- You can define the bump type, using the `BUMP_TYPE` environment variable. The default value is `minor`, but you can also use `major` or `patch` (the script will fail if the value is not one of these three):
+
+        BUMP_TYPE="major" ./scripts/release.sh
+
+- The script will create a commit in the **main** branch.
+- The script will push the git the main branch including the tags to the upstream repository, https://github.com/testcontainers/testcontainers-go
```

Comment:
Let's switch those 2 lines around, since the tag push happens before the commit of the upcoming version.

Reply:
L26 is pushing the git tags, not creating them